e2e/test/scenarios/question/question-management.cy.spec.js

```diff
@@ -178,19 +180,100 @@ describe("managing question from the question's details sidebar", () => {
               cy.findByText("This question has been archived");
             });
 
-            it("should be able to add question to dashboard", () => {
-              openQuestionActions();
-              cy.findByTestId("add-to-dashboard-button").click();
+            describe("Add to Dashboard", () => {
+              it("should be able to add question to dashboard", () => {
+                openQuestionActions();
+                cy.findByTestId("add-to-dashboard-button").click();
+
+                modal().as("modal").findByText("Orders in a dashboard").click();
+
+                cy.get("@modal").should("not.exist");
+                // By default, the dashboard contains one question
+                // After we add a new one, we check there are two questions now
+                cy.get(".DashCard").should("have.length", 2);
+              });
+
+              it("should preselect the most recently visited dashboard", () => {
+                openQuestionActions();
+                cy.findByTestId("add-to-dashboard-button").click();
+
+                findSelectedItem().should("not.exist");
+
+                // before visiting the dashboard, we don't have any history
+                visitDashboard(1);
+
+                visitQuestion(ORDERS_QUESTION_ID);
+
+                openQuestionActions();
+                cy.findByTestId("add-to-dashboard-button").click();
+
+                findSelectedItem().should("have.text", "Orders in a dashboard");
+              });
+
+              onlyOn(user === "normal", () => {
+                it("should handle lost access", () => {
+                  cy.intercept(
+                    "GET",
+                    "/api/activity/most_recently_viewed_dashboard",
+                  ).as("mostRecentlyViewedDashboard");
+
+                  openQuestionActions();
+                  cy.findByTestId("add-to-dashboard-button").click();
+
+                  cy.wait("@mostRecentlyViewedDashboard").then(
+                    ({ response }) => {
+                      // no recently viewed dashboard
+                      expect(response.statusCode).to.eq(204);
+                    },
+                  );
+
+                  findSelectedItem().should("not.exist");
+
+                  // before visiting the dashboard, we don't have any history
+                  visitDashboard(1);
+                  visitQuestion(ORDERS_QUESTION_ID);
+
+                  openQuestionActions();
+                  cy.findByTestId("add-to-dashboard-button").click();
+
+                  cy.wait("@mostRecentlyViewedDashboard").then(
+                    ({ response }) => {
+                      expect(response.statusCode).to.eq(200);
+                    },
+                  );
 
-              cy.get(".Modal")
-                .as("modal")
-                .findByText("Orders in a dashboard")
-                .click();
+                  findSelectedItem().should(
```

Comment:
We should probably assert that it's possible to save the dialog with the pre-selected item if it's not checked in another test

Reply:
it's is checked in the previous tests, so I didn't want to cover the same several times